### PR TITLE
Add substeps to the directions

### PR DESCRIPTION
**Pull request by andreeaceachir142**
# What 
- Add substeps to the directions
- Integrate the functionality with the final step
- Create a new component for the `route-instructions-maneuver` with the new design, and rename the old one to `route-instructions-maneuver-legacy` 
- Make sure the `route-instructions-step-legacy` uses the `legacy` version of the maneuver
- Boy Scouting

# How 
- Create a React wrapper component for the `route-instructions-step` and use the `useImperativeHandle` in order to trigger methods on the component 
- Adjust the `Modal` and the `Bottom sheet` components to contain the substeps and display them correctly 
- Set the steps of the final step to be `null`, so no maneuvers are shown 
- Handle swiping gestures on the bottom sheet - open the substeps when the user swipes (opens) the bottom sheet completely, and close the substeps when the user swipes down
- Add a `resetSubsteps` function that handles the closing of the substeps and resetting to the first step
- Add new icons to for the directions arrows 
- Add event emitter for toggling the substeps on the `route-instructions-step` component

### CODE REVIEW

File: packages/components/src/components/route-instructions-step/route-instructions-step.tsx [hunk 1]
```diff
@@ -283,6 +303,10 @@ export class RouteInstructionsStep implements ComponentInterface {
                 }
                 <div part="step-heading" class="step__heading">{this.getStepHeading()}</div>
             </div>
+            <div class="step__details">
+                {this.renderToggleButton()}
+                {this.renderSubsteps()}
+            </div>
```
[the remark this answers is not in the document]
Comment (andreeaceachir142) [reply packages/components/src/components/route-instructions-step/route-instructions-step.tsx]: Good idea. Refactored the code into a separate reusable function. 👍🏻

File: packages/components/src/components/route-instructions-step/route-instructions-step.tsx [hunk 2]
```diff
@@ -92,6 +92,27 @@ export class RouteInstructionsStep implements ComponentInterface {
...
+     * Event emitted when substeps are toggled.
+     */
+    @Event() substepsToggled: EventEmitter<void>;
+
+    /**
+     * Programmatically open the substeps.
+     */
+    @Method()
+    openSubsteps(): void {
+        this.substepsAreOpen = true;
```
[the remark this answers is not in the document]
Comment (andreeaceachir142) [reply packages/components/src/components/route-instructions-step/route-instructions-step.tsx:105]: Which event are you referring to?

File: packages/components/src/components/icon/icon.scss [hunk 3]
```diff
@@ -101,45 +101,45 @@
...
-    mask-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill-rule='evenodd' clip-rule='evenodd' d='M16.8945 8.61398C17.5057 9.4416 15.3154 10.0565 14.0492 8.99166H13.9937V22.0005H9.89966V9.03224C8.62687 10.0405 6.50226 9.43087 7.10551 8.61398L11.5577 2.23569C11.7946 1.92209 12.2055 1.92209 12.4438 2.23569L16.8945 8.61398Z' fill='black'/%3E%3C/svg%3E%0A");
+    -webkit-mask: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill-rule='evenodd' clip-rule='evenodd' d='M12.0001 22.5C11.4478 22.5 11.0001 22.0523 11.0001 21.5L11.0001 11.5L7.9586 11.5C7.1888 11.5 6.70768 10.6667 7.09258 10L11.134 3C11.5189 2.33333 12.4812 2.33333 12.8661 3L16.9075 10C17.2924 10.6667 16.8113 11.5 16.0415 11.5H13.0001L13.0001 21.5C13.0001 22.0523 12.5523 22.5 12.0001 22.5Z' fill='%234B5563'/%3E%3C/svg%3E%0A");
```
[the remark this answers is not in the document]
Comment (andreeaceachir142) [reply packages/components/src/components/icon/icon.scss:104]: Yes 👍🏻 UX approved

[the remark this answers is not in the document]
Comment (ammapspeople) [reply packages/components/src/components/icon/icon.scss:104]: 👍

File: packages/components/src/components/route-instructions-step/route-instructions-step.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ammapspeople) [reply packages/components/src/components/route-instructions-step/route-instructions-step.tsx:105]: The substepsToggles event.

**Review by ammapspeople — APPROVED** [review]

**Pull request merged**